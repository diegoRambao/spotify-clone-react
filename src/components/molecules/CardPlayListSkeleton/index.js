import { Skeleton } from 'components/atoms/Skeleton'

export function CardPlaylistSkeleton () {
  return (
    <div>
      <Skeleton variant='rect' height='8rem' />
      <Skeleton variant='text' />
      <Skeleton variant='text' width='6rem' />
    </div>
  )
}
