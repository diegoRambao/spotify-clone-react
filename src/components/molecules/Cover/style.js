import styled from 'styled-components'
import { LazyLoadImage } from 'react-lazy-load-image-component'
import 'react-lazy-load-image-component/src/effects/blur.css'

export const Container = styled.div`
    display: flex;
    align-items: center;
    gap: 1rem;
`

export const Img = styled(LazyLoadImage)`
    width:50px;
    height: 50px;
    object-fit: contain;
    border-radius: 4px;
`

export const Title = styled.h3`
    margin: 0;
    font-size: .9rem;
    margin-bottom: 5px;
`

export const SubTitle = styled.h5`
    margin: 0;
    font-weight: 400;
    font-size: .8rem;
`
