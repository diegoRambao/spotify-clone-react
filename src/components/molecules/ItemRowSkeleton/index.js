import { Skeleton, Box } from 'components/atoms'

export function ItemRowSkeleton () {
  return (
    <Box className='d-flex justify-content-between'>
      <Box className='d-flex'>
        <Skeleton variant='rect' width='3rem' height='3rem' />
        <Box ml='.5rem' className='d-flex flex-column'>
          <Skeleton variant='text' width='4.5rem' height='.5rem' />
          <Skeleton variant='text' width='3rem' height='.5rem' />
        </Box>
      </Box>
      <Skeleton variant='text' width='3rem' />
      <Skeleton variant='text' width='3rem' />
    </Box>
  )
}
