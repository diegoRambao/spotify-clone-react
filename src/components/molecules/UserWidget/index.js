import styled from 'styled-components'
import { Avatar } from 'components/atoms/Avatar'
import { firstLeetterToUpperCase } from 'utils'

const Container = styled.div`
    display: flex;
    gap: 8px;
    align-items: center;
`

const Title = styled.span`
    white-space: nowrap;
    overflow: hidden;
    text-overflow: ellipsis;
    width: 200px;

    @media only screen and (min-width: 768px) {
        width: 130px;
    }
`

export function UserWidget ({ name = ' ', img }) {
  const firstLetter = firstLeetterToUpperCase(name)
  return (
    <Container>
      {img === null
        ? <Avatar>{firstLetter}</Avatar>
        : <Avatar src={img} alt={name} />}

      <Title>{name}</Title>
    </Container>
  )
}
