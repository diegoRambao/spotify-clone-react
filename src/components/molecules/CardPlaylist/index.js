import { Link, CardContent, CardImg, CardBody, CardTitle, CardSubtitle } from './style'

export function CardPlaylist () {
  return (
    <Link to='/'>
      <CardContent>
        <CardImg
          src='data:image/jpeg;base64,/9j/4AAQSkZJRgABAQAAAQABAAD/2wCEAAoHCBYWFRgVFRYZGRgaGhocGhocHBocGhocISEhGh4cIR4cIS4lIR4rIRwhJjgmKy8xNTU1GiQ7QDs0Py40NTEBDAwMEA8QHxISHzQsJSw0NDY2NjQ0NjY0NDQ0NDQ0NDQ0PTQ0NDQ0NDQ0NDQ0NDQ0NDQ0NDQ0NDQ0NDQ0NDQ0NP/AABEIAOEA4QMBIgACEQEDEQH/xAAbAAABBQEBAAAAAAAAAAAAAAAEAAECAwUGB//EAEEQAAEDAQQFCAcHAwUBAQAAAAEAAhEDBBIhMQVBUWFxBiIygZGhsdETFoKSssHwIzM0QmJy0hQkUhVTouHxg2P/xAAaAQADAQEBAQAAAAAAAAAAAAABAgMEAAUG/8QALxEAAwACAgEDAwMDAwUAAAAAAAECAxESITEEQVETInEyYZEUgbEFUvEjQqHR4f/aAAwDAQACEQMRAD8A7/TWmH0qlxt2LoOIM4zv3LP9ZKuxnYfNV8q/v/Yb4uVFg0SajL98NxIAIk4a8/qEdJLbM9VbppBXrLV2M7D5pHlLV2M90+aDraJe2oKTTeJaHTEBokjHHd3qrStkFJ4a197CTtB38c/qUUkxXWRLbNH1lq7Gdh80x5S1f0e6fNYV5KUeCF+pfybvrLV2M90+aXrLV2M90+awryUruKO+pfybp5TVdjPdPml6y1f0e6fNYUpl3FA+pfyb3rLW2M90+ab1mrfo90/yWFKaUeKO538m+OU1X9Hun+Sb1mrbGe6fNYMpLuKByv5N71mrbGe6fNL1nrfo90+awZSlHgjuV/JvDlNW2M7D5pvWetsZ7p81hEppXcEHlfyb3rPW2M90+aXrNW/R7p/ksElIlHgjt38m8eU1bYz3T5pjymrbGe6fNYJKa8u4IP3v3N/1nrbGe6fNN60VtjPdPmufvJXkeCKKb+Tf9aK2xnunzT+tFb/8/dPmuelWUaZeYkNABLnHIAd+6BrK7ghljp+5uHlTW2M90+aK0Tyhq1KzKbrkOJmGkHInbuXJkrS5Ofiaf7j8Ll1RKllZxP5PRUkklnG0cZyr+/8A/m3xKybMOez9zfELW5Vj7f2G+LllUXBrmk6iD3yU6fRjv9bOzqvBL2MIFS5IJGrENO8Az271xVVrg435vAmZznXO9bNotTn1xUoAuuMEjKRJkY8R9BUaV0qKrWgMuwZknHZGGryCWdopkaa8/wD0ykoTgJQn2QGhMnTtR2BkU0qUJQjsAwSSKSIRilCcJQu2dsaE0Kd1K6js7ZEhMpFqYBdsPIYpEJFqQCOwpkVJji0ggwRrw4a0iFZZaIc8NJDW6ySBhsBJiTkOpHY80Ef1T2sJcQS8ENENyyc/LqG+9sVdV5pQxkB0NL3wC6SAboJyABjDMg7oe00nOJcSwDU0VKZgAQGgXpMDt4pVW+khwLQ+GhzXODZgXQ4F0AyAJEzM5oLRZURZU9Jea+C665zHwAZaC66SOkCAc8QYVzG1PRs9GMCHTg3O87aNkKljRTBcXNLy1zWtaQ6Lwulxc2RkTAmZ4JOpX2U4czAOBDnsaRzycnOByXMrNAdeZN7pSZyz6locnB/c0/3H4SgHsgxswwII7RgRwWhyd/E0/wBx+FyZ/pY6r7dI9ESSSWQU47lSPt/Yb4lCaN0h6K9zA69GuIidx2ozlT9/7DfErKs9G+9rMrxAknLf2I+xlravaOrs9tBY172hl9wAEzmYGoceCF0hpU0nlnowRAIMxI4RtlR0hTY91wvDGUmi9tkjmxwA74Q+lYqUWVRmMDqnUe8d6VJFab0/2My32r0j75bdwAiZ70MkQnTozPvtkYShPCUI7F0MmIUxOaaF2wNdDJXVaxhJCJpWYrnWh5x1XgEDNykKJ+uKKqPYzM47B9QgbRp+mzJo7Urv3G+nKem+/wBuy/0B7kjZzmserytIPNYFR63v/wAB3eST60r3H+gn4TN59Eqt1JZtLlePzs7CtOzaZs9TC9dcdRXTlT8MDwJft+SosUIWm6ygiWkHYRl3fWKFfRIzVlZFxU+QYhWWcMk39mHHCJjVtUXDYokKm9gVBDRTjEmbh29PAjLViR1b0rtLm4u/WAMchgOLgTwduQwCS4oqL2Npw2STneiZIl23ATzYjfKTG0oF4kmROBAi9nt6OPUq6OAcYEhoiQD+YDXxVz2PDA+GwSR0WSI3RP0NqYoqBKgEmMpMROXXj2o7k7+Jp8T8JQloHOOWo4AawCcEbye/E0uJ+Ern+llJo9CSSSWUochyo++9hviUNYPQXT6aSZw6eXs7/BF8px9t7DfErHa0EgHDUTs3obJNfds37RZrMy6XsIvYjF+7Yd6D0vZKbWMewEB28nCARmVs2mnTLmB8EmQwGYOU7tQzWNp57y8Nc2GN6JGR/wC8MtSCY1JaZlQmu5p05R2R4kIThqeE4am2JxGKmxh2J2U0TIY2T2Jdloxf91eBANYJd2ays63aROQwGxKrULjJQ77PIRUsyZvVbfGel/kyLXWc5ZzqZ1rcqWRB1KGanWNvyNhzSlpGS6mqyxaNRgQ7mqLxm2MrYKWJoRJaq3NSONFlew7RmmalIjnEt2LsbBpBlcbHdx8l56Wq+y13MMgwqRka6ZO491/HszvbRQuoQtV2iNJiqy6/B2U/JTtFKFqmjFcLW1/wCEJoUyEyqqJqi+x4XzruO8EXVFxtIhz7zhLg4bT0pkxOEDdOtB2cgB8/4EYDaQPmi7XaDDA6o5zSb8DEbIxeYAiI3Fc32VmujPtfTPV4BF8nh/c0+J+EoW0jnHg3wCM5Pj+5p8T8JTb+06a+5HfpJJKBsOS5Tj7b2G+JWfZrG98ljZAwOIGPWVo8pvvvYb4lD6IruD2sDiGkkkbTH/QSNncdhVWw1qsX7rCzowc5jYTsUbay0PYGvYIbiSC2TAOYB8Fq0Xk1KjScAGQNkgygdGWh76dS+68QDGWw7EAuTACZSSXE3JCFcymmY1EHmjeg6UrsrjxJ90TawDEntyWfarUwnF2H129SG0nagMCZKwatYpsWPLk7S0v3L3hVTp9L/JqV9JNHRg74J8oQlTSjtTiOAb8wVlucVKhZXv6LHHeBh2rfHpVK3TMn9Nj3qZ2Gm3uP5nd3kmFec3Hu8lR/QvGbYjeFF1NzTBCqsMAeBT5nX9glwaZnq3KBsjTk7uVIepsqIV6aX7A4r2Hdo1xyc09Z8lU/RdTU0H2h80XTqq8EkYPI7CO8LBnwzC2FKkY7tG1R+Q9RB8CoiyvHSY4cWnyXTU6uGJGOvgiK1VwbcZmROOAA2krFPF9iVmpdaMKwNcxwOorrqFW+zHpDvC5Y2pkS1wMHZzT5dWC19F2sYOacNfzC1ypc9EFVTe6XT6YRVZjkqkda6cHDYg00voXJPGtMTHASIkERnBzB+XenL2n8rveGr2VGEybYmxVHSZ+oAhHcnx/cU+J+EoAhaGgB/cU/3H4SjvoaH9y/J3qSSSkbzk+U333sN8SgtGPAqsJIABOZwGBRnKZ323sN8SsqniQN/YpPyaJncnQ0rbTD6ji9sG5B2wDMIOwV2Movl3PdIicThAw2Yo6waPY032Pv5jMEDblr81maVsjGGWvlxdi3DDXkMhiuF0m9GcE4Ca6psQ2Mo2wizsnE5DFCW61SYGtEWmpcZhm7wWXZpmTlqV8WJV9zNGplbfsZVspuL3dXgqf6QldDaaLSQ4EGRjxVJphejjfSPP8AUeqe9IF0bYmtF9zbzpwnIDbx8kbWtD4iQ3gFFtUNEIW21fBI5dV2NPqaWNcQOs5xMkntQzwrHOlDuM4BaUkhFd12yDkmp3MhTpsXNrRSYe+yyiEU9xDcNhzwGOWKFZWAcWmRhry2zKnaC5rQQcsMsMA7VmvF9bnVPiijlIufVN5jcQCG4RtP13Kekqh9C/e0weGB7h3oa0NdfpgCBdbnwB7d4RxqBwAO1wx1zjl1rHinc7MXqK43s5yy85hbsGMZxMytmx2hoqU7uVSm2R+tst8GoZmibry5rgGY7SQDqjWQp0KDRaKIafylx39L5q0Kpe/wPkrHctJ+x17OcwbRh9d6EeMUVZtY3KiqIMK6enonknlCplMJEKV1RAT7M2hoR+gR/cU+J+EoIo7QP4inxPwldsM/qR3aSSSU3HH8qfvx+xviVk0mOc4NbmTA61tcph9t7DfEoXRdVjH3nNJMQ2IwJw19nWVJvs1qK4po0bVfZco0WlxbDnHIYGQCchJk9io5Q0Yc18Z4HiMR3eCru2mXkNcC8ycPDZhh1BV2l9UUgx7ebPSdMzic54oNgUtPezOBKIos1dSoai7JmTsBP11oGjGtdg9vdLo1DAdSpZACrr1MSqX14XoQtJIpWF0tBD3CELUriCqX1TPFBEkzK0TrRgr0b5bLKlqOQQ9prSVJjCASqBTVVS2N/SuZ18jUzj3KbqJBwnNFWWz6zkmZJfdEw3P59JZ8vqYh9s0R6eZn7gUNMOkER38EZRoABs6/KfBRcy+DdMDpTMxqGXX9BSY9oDTgCTOBwETlOvUYXn5vXN9SJdTPgDtHTcQ4ANAGrnZAfW5W1qr3mAMcdfVqic+5KzEXgLoxcO7ZvVtCnedlgDOqY6eevEBedT29szum+yhzvtGyehzdu0R3Igklrv0wQcjERkqKLefOfOJxz1N8yigyXY5EQOPlB7lohaWjFm7rZXUtXMcB0xmPragnVLtW8T0W3Z4ZkcTPajabIlpGLcAdeGQ3xqO9ZltZdgbWnxhF5F4RXFgal010zubMcQdoHf8A+qFoGKhop96nTdta1WVsyrctvY3H/pa/coISShIBPsycRiNyO0D+Ip8T8JQSP0GP7inxPwldsMz9yO4SSSRNZynKX772G+JWUx8EEZjEdWK1eUn33sN8SqLDo8PbeLw3EiIHbmoV2z0oaWNNh+lrY9ly4QJBnAHKPNVaVeXWdjnZktJ4lpRNusrH3L1QANmYjHLacMt6z9K2prg1jOizXq2ADcBrXNkolVpJfkzAjKDeY7eY7EJC0aLOY0bcVOr4LkaK+1GRUsxKpdYyt0U1I0ly/wBRSKzmaXg5+pYS6e7duQ39EV0VoqsZ03tb+5wHisq06aszB94HHYzneGC5f6jT/SmwPOl50Uf0JLRuSp2HciLHpJtSm57AGiSG38ZI2gERntWfX0hUdgyq2ZE3WAdUuvYqNety1tb0Nu7naQtIgNA3GSMY4EBYrqxJccpnbsjtglaNsxb6R2JnMxn4LGtFrDsI7EMbd90zLm2lrYay2BoN6JII6p2akO63MJBOoRDRCBeC4ye5OKfUqqPhGJ692Gs0kGkG6TGOzb5qX+quIgMA1ZnsQlOjPzKk8DIfW9UWF/qYrqfBOlbH3jESY1ZRMR2lEivUvNcTMEEtgC8NmAQlGoGnL628EdStzARExrwQpaQZmXW9aDqFOGidkoS10C94azE5bsVKppRgGAJPUPBVvtLvRPqN5t1zYIz+p+ay8a3s2Vc8NbOzs1lFNjGDJrQJ4a1TV/8AEZe5gJzIB7v+0DUzWyGQqdY/yyuEgFI5qJT7MjgRR2gx/cU+J8CgSFoaE+/ZxPwlMmKp7O2SSSRLaOT5Sffew3xKpsVkpubL6ga6ThLR44q/lJ997LfmsponDMnvUKf3M9SJdY1p6N+yaMpYkG+DhmCB2a1naTsjGRdfLpgiRgInIY7EdabzG06TBLhziBsGJ6ifBD6apg3KjcnAeY7vBM9aJ43XNbfTMtEV7fcDWsY55ujKA3tz7kMFMFSqVS0zVWPkvIDVt9tcYZSpsG1zrx8R4IHS9W0Bs1XXnPkMp03PE4Y81gyGcnhrXQMxKDrWKu95c2o1jchcabxaMRLj4DBSWKZXSMueJhLttnDVtF1m4vpvBO0QexU/0j/8T2Fdw/RzgefUe4n9RHzVb7C2IJd2rRjwNrto8zJ6ty9JM53R2lfQsLHU75Ly4SboEgDKN0qi16Ue/CGsGxo8SZ+S1bWyizUXHZOCy32kflYG8B8zilfpZl7bWyuP12Sp0k9AnoHES6Y3q1lMJzU6+KY1TtVJmJBVXXkncjPBM54GWKoLkgUeeukBR8lrqxVRKdIMOxBt0FJISSkGjWQO9M6o0ap4pXPu2FP4K3ulbVg0ZaXtAuQyb0PhocdRI6RVGgqd+q0mIbJ68h2TPUu1ZUwB2ZpJnbbLxM11XgrsrHNZD4vHEwZHUSk5I1M4yzCoLiqKRstSpUyi0hMVWHFSDkxmbTJI/Qn4inxPwlZ5cUdoP8QzifhK5eQNdncJJJKoDlOUf33st8Sg7HReSHsbeunqntRvKP772G+JVOi7S4PYwOhpJwgbJziVnf6j1p39BNfHuFMbaHF5hrC4AScCI/xie/ahLQaraQY9kNB6RxOcjXgtylUJe9pyaGR1gygK9Uvs15xkkiT7cakzXRCKfJbS8owk8pEJQpHoaL7P0h9akPQ08wXpLGtum4XGXOO0NGQ2TmraLoI2fJcYbJctNRjwCR0ZGBbtHckvwZPUp8loMtemHueHzIbIBAujHXjwWdaLc9xxce0rTttEXCcrscJOrsHcuee5PNvimjyaxatp9knvlQvKF5PMoqtleGh7yiUnOAVlloOqODGkCdpj6KFVpdlJh09JFd4JB60q+h3MGLXE7SMFmEQSkm1XhlbwVK7J3imLymbVh0jLEdRT0BL2g7VXl7JkXOhnNKiRCMrtAeWj6PkoWhsNbIz8PrwXVK7Z0tmloSm1xLhzXNEg7Djq1jdvXTWfFrSQJIBxXK6Dok1SwdFzedvAIJHy6yuurAE8PCEIfktEbTogBjn1p7qiJJhJPQy1rwOYTEJJwl2K5TYpR+hD/cU+J+ErPR+g/wART4n4Sml9k6jpncJJJKxA5DlPUivH6G+JQWiq49KyTAk54aijOVDZrj9jfEoCxWcF4LmOcwHnXQT1YKT1yNk/V4deNHQUbWwVKpL2RFPG8IOBmFm2m0sZZmMvAucGmJxEm+Z2bEfaLLQZE0SZ/wAWudHGDgsrStOmQ25TLMTJc0tnLai9LyRnlT689f8AgFY6QpKhohTL1FnrRX2/cWtKzeUFI3G1g0PNPpNM5ZXgWkHUARrEbEWKwV7HTwK5r5EtTllpM4i3aUc8XYDWjJoy47zvKzHO1ra0noxlN5vGoGnFt1gcDuDi8dh70NZtH1Xn7Kk4CcHv8ZIAHsgnep/seasdJ+OzOu7cPE9Sk5rrocGkNJi9jiePkux0ZyYptF+qfSPwJaegPm48exAcoXsNVrPyMY5zwMucSABsdAbG8hT+quXGS6w9boF0PYGPpvaWn0rWl0HWw5OaNoOf1GRTYWvLdcx5FaOjKzmPZUaTIABLQTicJcJxbjiNqK0ZZgbU11UAXnSGiY445Nx17EHWtlZnaWvZmpo9lcM555kZOz6tywdLMFynAxffeeBJI8V2un7UxjCxgvVHjmicm63E5NA27Vxlosr/AE7PSC6A28MDdutxgTmBkox1W2zRy5LwA2my3A2QZIxmO4bPJUg3SCMwQVpVWF7rzsgHO4AYAfNZtpGMfWOK1RfRlzY0nteArAvD/wAsAndGEcVTUffmeIVnoHNYAcC4kncAbvjJTWazuc+6dUzu+iE/Log4fg1+TtRoqwTBLTHiV0uY3yuOYGirSIdADhJ2CQMl2FWiRln9Ep8fuVjal9b0Su4prqGl0qDqzlRY2/Av9VHvLCy1RhDMrGUTeS1DnpjRlm96WhoWhoP7+nxPwlZr3FG8n3zaKfE/CV0S/JO8iTc/sd4kkkrmQ5LlMftvYb4lBWTSL6YIZGJnETu+SN5Tj7b2G+JWOAsz6pnqY0njSa30dNpC01W3DTbeBGMNLtkZLGt7q9SLzHc2YhjhnHklZ9JVGNDWnAZAiY69im/S9UgiW4gjJM6RJYqXSS/PuZzTtTKUJAJDRx9tkbiuYotCotVuZT6RxgmNeHyXNjSpjthpqQDjGGB2HUgKxxl9R7nf4tAYOBIkkcChbNfqEPe4Bt0kMG04YytCnZ5cTEw0DdlAWPNm09Scnze30gapbnBpawc4znMNAGZOoeS56zsbUcA4g33EyP0xhrN2ANnitE1nufUYyL75a0GIAOGIOHQElV6PDKbiA1pe2A6oA6C45hoJyAGcY55QpSuMt+4X916XgrtdKo18MzAwLczMtAMn8oHz14SsdmcXhjA0PAMEybrBiIGoTvxnejbcHPZfvXWBtRrmyRLoJa4wdsKmxvayzvqMHPLGM3jmgkTuELlTcj61Rr2GxMvuc833uILidZ1HgNQyHerNJMD3XSJAvat0LHslc8x4JhoAfqB1Az39a1aFqBLetp4Yz14qFckPw09o5yy0i51UERevRdzwcRr15IK1WO7DxzmlrshrESDjgROS6CvRLKt4iWHOPyE4E/tMgypV7ELj7rgGvN6DiA8eeKtObT38grGqnRh+jllNwkwHNdwcMD2yDsgqdQXGueAQC2BtacojqVlKjdiXC7JaPaAx6nY9Z2q2uy7SF9pIAE7WiMezHDsVlXZHhqWYD3ZY5Y79qLpabrtODzhqdzhwgoSrUgXG7cdc6hjExn2qr0a1d+x5++L3s6SxcqMYqsH7mfxPyK3rPUp1BLHB2WWYnKQcQvPLuxWelwulo3HEEbp2bkU37lFna6aTR6C6zgGYSLVwdDSFRhlj3jcTI6wcCtyw8pgcKzQNjmgxwLcT1ou9+SsXj9lo3ijdBfiKfE/CVgWPSYqvhjDcDZLjhB1NjWt7QX4inxPwlND2yWTTR3SSSS0GM5LlJRea0tIi43OdpWZTslRxiW8cVu6d+89kfNC2U4leJl9ZknO5Wtb+DZFNQv8A2CVLEWjHHrQr7O8ZAEHLGF09UMe0B04A5bcFn2gAAAJs3qskdrTWjptv3ZimnU/xB4ELPtWlWsMdI5QzndWGHeta30y4NbMMJ55mObBOG3EARrT2GxgXgWi/DQ4tiHtBJDgd4OWozuTYfVVc7pL+w+6fhgDLNXqtvSaTdWRefJZ9lsjw98HWMSJJjAGc/wAs+0uhtda+64DgDBA1lZdvq1GseaZIawYY4OxJPVgRHFG7bGmeT7CQYIDsHOcJGAhoynqGW9FioGMwzdlvOcrjrA95e+q8nm0XuaCcSYDAcTnLhidm5S0XbnAOLi57mtJxOAGcknh3b1CsVPvZdNLph1jpObWqvI5rRzv3ughg6gJ48FU4EMDiAXlxhuwknE7Nm+Fp1ntbTvtbDnuvAOwAc5o5x1wACRuMDFYWlahN5jGl127fJ1nUIy6uPFGd0x5amdsofpF5a5ubS8kbCA0gtG3V3Imx2pvo7mOL2YnI8y6QBwb3oCm11/U4N5ou4hovZCDBcdmOeK2LTTIY590B7QWgAyCcyccyOjOu6qVxWkLjdVTYfScym03ug8wRiSBv2DArJtmkCXOazESJIyA2g7TKt0RZnuYWvMYXqf6XR0eDmjqz1oe0aNult0EYkG7E7Jx44jwU5mVT29leVa2kWWfS1T0cvJcWmAYuuwznURH/AKrKulC+m83ACS1sgwHOx1CAOKBa6Xmk1swIOwHDnN1zJPVAVttpgObRYCGskvjGXHG7OUgb/wAvUqcZ34/f+wnLS6/H9ykvIeJHNDZwjUyTB2YK+21g+nDXk3hiScCSTGeWIjdwRVjsjXAuEEvD2gDo5O255QsSwuHQfIbInbmL3XGPsgcWhKn+CeSnPXzsHFMziNk/XUrQCcAk5jr7uMnZjjPA5qTntbmSTuwXqY5Snb6PIyN8tIcUh/6oehJwAPYqnWk/lHbiVFznnMmO5Cqh9JN/gWZpdt6LWWJ73XWNLjsbzo4xgOtbdg5Lu6VYwP8ABpknich1IPQmmX0OYReYTiNYO0b/ABXWMtReA5jS5pyOpZ2p8vr8m7D9NL7nt/BBlBrGhrGhoGoI/QX4inxPwlBPa8Ylh6sfBF6Bef6imC0jE6v0lPDl+Gn+GjstbT0n/B3iSSSvoxHIcp65bWgOjmNwgHWVlMtVQ9GDGwYrp9NaPc+peDLwugThv2rOOhn/AO18PmvNy3HN7xN9+deTRLfFaaQEdJPu4s50HZGUk57ld6QkXnYccFf/AKK/L0Xw+amNE1AIFP4fNYs8zc6nHSey6qfdoEcQQZxGtUWWocKbjjF5p1tn8vfC0/8ATKv+B7W+ai7RNQ/kPEEA9sqWCcmN9y9fgV0t7TMxgxiIc0P7Ddg9hVOkaF2i5owL9+3M989S1ToN+BuvlswS6SJzzKVTQb3GXMeT+/yK0ZHTe1L/AIK480T5ObZZ4a4Z3qZA4NLXDuCC9CG2Z7Bg6pUgGMSAJIHGCPaXXHQD8JY/DLn5f8setRfybJgGm4xMc84Tgfza101a8y/4Hyeox0ujmHF91z3Ak4NpiRzWiGAjge2DxTUKbWWdz2Plzw087Egnd2cMV1DuTROdN3vn+Sg3kqBlSd75PcXQuVV/tf8ABz9RBy+hdEEPa94ht3bk84Ybm59Yz1aFqZgG+yPD5roToWocC18REXgMOohQfoB5zY8+2f5JKd1W3L/gMeoifBl0LPeLGxEAuI2flA7FKtZibwbm0HtIWo3QdQEkNeCcCb/mYTnQlTHmvxz5+fel43/tf8B/qp2cwyz3CXgZNbdG10Az3dwWEWX67XgRILjiTjJBjdkAu+qcmi6L1N5gyOe7A+8q28k2gyKLpmem7P3laKqe3L3+BK9RD1+Tl/SkMMOaHtfIB3ROrZPas61PY17jdBOQxORHeYOEZRns7d3JBpMmiT7bv5qJ5Gs/2Dh+t38lSMql74V/BLLm5dLRwD3l7GgCHNwO9v5TxBkcC3Yr6Gi8LzzAzK7lnJBgyoEe2f5Kb+SgOBouI2F7v5LXHq4nusdN/jox3NU+ml/k5KxWNtSk97BduOAGGYOs75VYokAhy7WjyaLGlrKTmtdmA8wf+SrPJNv+yffd/JWj/UuKe8dfwSv0/LWqR5/amZwui0LaCy605FrZG+MStz1RZ/sf8j/JW+q4kH0JkRHOOEZfmWfL6qMicvHWn+w30aWtUuiQKL0X96zifAqLdF1AAAwwMBiPNFaPsNRtRjnNIAJkyNh3rx8OHIsqenra9mbKtcX2dCkkkvpdGTYkkkkWASSSSDOEkkkiwiSSSQOEnSSXHDJ0klwRJkkkQCSSSQOEkkkuOHTJJLjh0ySS4HuJOkkiEQTJJLjh0xSSQRwkkklUB//Z'
          alt='Poster de Karmadome'
        />
        <CardBody>
          <CardTitle>Karmadome</CardTitle>
          <CardSubtitle>14 canciones</CardSubtitle>
        </CardBody>
      </CardContent>
    </Link>
  )
}
