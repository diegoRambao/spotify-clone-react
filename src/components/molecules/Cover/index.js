import { Container, Img, Title, SubTitle } from './style'
import PropTypes from 'prop-types'

export function Cover ({ img, title, subtitle }) {
  return (
    <Container>
      <Img src={img} alt={title} />
      <div>
        <Title>{title}</Title>
        <SubTitle>{subtitle}</SubTitle>
      </div>
    </Container>
  )
}

Cover.propTypes = {
  img: PropTypes.string.isRequired,
  title: PropTypes.string.isRequired,
  subtitle: PropTypes.string.isRequired
}
