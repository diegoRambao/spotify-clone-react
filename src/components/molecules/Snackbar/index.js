// import { useState, useEffect } from 'react'
import { Container, Button, ContainerWith } from './style'
import { Box } from 'components/atoms'
import { CgClose } from 'react-icons/cg'
import { BsCheckCircle } from 'react-icons/bs'
import { BiErrorCircle } from 'react-icons/bi'
import { IoMdWarning } from 'react-icons/io'
import { AiOutlineInfoCircle } from 'react-icons/ai'

const iconsSnackSeverity = {
  error: <BiErrorCircle color='#fff' />,
  warning: <IoMdWarning color='#fff' />,
  info: <AiOutlineInfoCircle color='#fff' />,
  success: <BsCheckCircle color='#fff' />
}

export function Snackbar ({ open, onClose, children, severity }) {
//   console.log(open)
//   const [show, setShow] = useState(open)

  //   useEffect(() => {
  //     setShow(open)
  //   }, [open])

  return (
    open && (
      <ContainerWith>
        <Container severity={severity}>
          <Box className='d-flex align-items-center'>
            {severity && iconsSnackSeverity[severity]}
            <p>{children}</p>
          </Box>
          <Button onClick={onClose}>
            <CgClose />
          </Button>
        </Container>
      </ContainerWith>
    )
  )
}
