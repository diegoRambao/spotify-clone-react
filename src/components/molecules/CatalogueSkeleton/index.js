import { Box, Skeleton } from 'components/atoms'
import { ItemRowSkeleton } from '../ItemRowSkeleton'

export function CatalogueSkeleton () {
  return (
    <>
      <Box className='d-flex'>
        <Skeleton width='10rem' height='10rem' variant='rect' />
        <Box ml='0.5rem' className='d-flex flex-column justify-content-center'>
          <Skeleton width='7rem' variant='text' />
          <Skeleton width='5rem' height='.5rem' variant='text' />
        </Box>
      </Box>
      <Box mt='2rem'>
        <ItemRowSkeleton />
        <ItemRowSkeleton />
      </Box>
    </>
  )
}
