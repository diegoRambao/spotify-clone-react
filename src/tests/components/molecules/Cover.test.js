import { shallow, mount } from 'enzyme'
import { Cover } from 'components/molecules/Cover'

describe('<Cover />', () => {
  const url = 'https://example.png'
  const title = 'Titulo'
  const subtitle = 'Subtitle'

  const wrapper = mount(
    <Cover
      img={url}
      title={title}
      subtitle={subtitle}
    />)

  test('render content', () => {
    const wrapper = shallow(
      <Cover
        img={url}
        title={title}
        subtitle={subtitle}
      />)
    expect(wrapper).toMatchSnapshot()
  })

  test('render title', () => {
    const h3 = wrapper.find('h3')
    expect(h3.text()).toBe(title)
  })

  test('render subtitle', () => {
    const h5 = wrapper.find('h5')
    expect(h5.text()).toBe(subtitle)
  })

  test('render img', () => {
    const img = wrapper.find('img')
    expect(img.prop('src')).toBe(url)
    expect(img.prop('alt')).toBe(title)
  })
})
