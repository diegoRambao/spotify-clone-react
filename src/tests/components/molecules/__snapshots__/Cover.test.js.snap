// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Cover /> render content 1`] = `
<styled.div>
  <styled.img
    alt="Titulo"
    src="exmaple.png"
  />
  <div>
    <styled.h3>
      Titulo
    </styled.h3>
    <styled.h5>
      Subtitle
    </styled.h5>
  </div>
</styled.div>
`;
